src/libcore/char.rs

```diff
@@ -176,6 +172,41 @@ pub unsafe fn from_u32_unchecked(i: u32) -> char {
     transmute(i)
 }
 
+#[stable(feature = "char_convert", since = "1.12.0")]
+impl From<char> for u32 {
+    #[inline]
+    fn from(c: char) -> Self {
+        c as u32
+    }
+}
+
+#[stable(feature = "char_convert", since = "1.12.0")]
+impl From<u8> for char {
+    #[inline]
+    fn from(i: u8) -> Self {
```

Comment:
I think it would be a good idea to add documentation to this method as it's not completely obvious how it's implemented. Specifically it should mention the code page.